### ISSUE TITLE
(aws-cdk.aws-synthetics): Automatic IROLE creation for Canary resource is not partition-aware for gov cloud regions.

### ISSUE THREAD

**Issue by timwelch** [aws/aws-cdk#12948]
When creating a Canary resource and allowing it to make the IROLE, it adds permissions for CloudWatch Logs but is not partition-aware so it fails in GovCloud

### Reproduction Steps

```
        const canary = new Canary(this, wId('Canary'), {
            runtime: new Runtime('syn-nodejs-2.1'),
            test: Test.custom({
                code: Code.fromInline(
                    Utils.getCanaryLambda(
                        `https://${targetApp.sslDomainName}${targetApp.healthPath}`
                    )
                ),
                handler: 'index.handler',
            }),
        });

```
### What did you expect to happen?

Expected the synthesized ROLE to contain aws-us-gov ...

```
        {
            "Action": [
                "logs:CreateLogStream",
                "logs:CreateLogGroup",
                "logs:PutLogEvents"
            ],
            "Resource": "arn:aws-us-gov:logs:::*",
            "Effect": "Allow"
        }

```

### What actually happened?

Does not consider partitions, and the CDK deploy fails when deploying in GovCloud.

```
        {
            "Action": [
                "logs:CreateLogStream",
                "logs:CreateLogGroup",
                "logs:PutLogEvents"
            ],
            "Resource": "arn:aws:logs:::*",
            "Effect": "Allow"
        }

```

CloudFormation Error:

```
Partition "aws" is not valid for resource "arn:aws:logs:::*". (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; 
```

### Environment

  - **CDK CLI Version  : 1.86.0 **
  - **Framework Version: typescript **
  - **Node.js Version: 15.3.0 ** 
  - **OS               : MacOs and amazon linux **
  - **Language (Version): 3.9.7 ** 

### Other

Please add partition awareness to the building of the resource string in the auto generated IROLE for Canary.

--- 

This is :bug: Bug Report

**Comment by NetaNir** [comment]
This should have been fixed in version 1.86 https://github.com/aws/aws-cdk/commit/c01272c14be9b7ff635281952f3cfeed971a352e. 

Can you verify you are using the latest version?

**Comment by timwelch** [comment]
You are correct! Our CDK container was apparently on an older version, even though Jenkins was supposed to be building it fresh every day. :-) Our fault, not yours! I'll close this ticket!
